crates/iroha_test_network/src/lib.rs

```diff
@@ -1,883 +1,802 @@
-                asset_definition: rose_definition_id.clone(),
-            },
-            ALICE_ID.clone(),
-        );
-        let grant_burn_rose_permission = Grant::account_permission(
-            CanBurnAssetWithDefinition {
-                asset_definition: rose_definition_id,
-            },
-            ALICE_ID.clone(),
-        );
-        let grant_unregister_any_peer_permission =
-            Grant::account_permission(CanUnregisterAnyPeer, ALICE_ID.clone());
-        let grant_unregister_any_role_permission =
-            Grant::account_permission(CanUnregisterAnyRole, ALICE_ID.clone());
-        let grant_unregister_wonderland_domain = Grant::account_permission(
-            CanUnregisterDomain {
-                domain: "wonderland".parse().unwrap(),
-            },
-            ALICE_ID.clone(),
-        );
-        let grant_upgrade_executor_permission =
-            Grant::account_permission(CanUpgradeExecutor, ALICE_ID.clone());
-        for isi in [
-            grant_mint_rose_permission,
-            grant_burn_rose_permission,
-            grant_unregister_any_peer_permission,
-            grant_unregister_any_role_permission,
-            grant_unregister_wonderland_domain,
-            grant_upgrade_executor_permission,
-        ] {
-            genesis.append_instruction(isi);
-        }
-
-        for isi in extra_isi.into_iter() {
-            genesis.append_instruction(isi);
-        }
-
-        let genesis_key_pair = SAMPLE_GENESIS_ACCOUNT_KEYPAIR.clone();
-        if &cfg.genesis.public_key != genesis_key_pair.public_key() {
-            panic!("`Config::test` expected to use SAMPLE_GENESIS_ACCOUNT_KEYPAIR");
-        }
-        genesis
-            .with_topology(topology)
-            .build_and_sign(&genesis_key_pair)
-            .expect("genesis should load fine")
+    /// Iterate over peers
+    pub fn peers(&self) -> impl Iterator<Item = &NetworkPeer> + '_ {
+        self.peers.iter()
     }
-}
 
-pub struct NetworkBuilder {
-    n_peers: u32,
-    port: Option<u16>,
-    config: Option<Config>,
-    /// Number of offline peers.
-    /// By default all peers are online.
-    offline_peers: Option<u32>,
-    /// Number of peers which will submit genesis.
-    /// By default only first peer submits genesis.
-    genesis_peers: Option<u32>,
-}
-
-impl NetworkBuilder {
-    pub fn new(n_peers: u32, port: Option<u16>) -> Self {
-        assert_ne!(n_peers, 0);
-        Self {
-            n_peers,
-            port,
-            config: None,
-            offline_peers: None,
-            genesis_peers: None,
-        }
+    /// Get the first peer in the network
+    pub fn peer(&self) -> &NetworkPeer {
+        self.peers.first().expect("there is always at least one")
     }
 
-    #[must_use]
-    pub fn with_config(mut self, config: Config) -> Self {
-        self.config = Some(config);
-        self
+    /// Get a random peer in the network
+    pub fn random_peer(&self) -> &NetworkPeer {
```

Comment:
```suggestion
    pub fn peer(&self) -> &NetworkPeer {
```

I'd remove the `fn peer` and rename this. IMO there is no need to get a specific/first peer, they should all be equal

Reply:
Makes sense.